actors/abi/cid.go

```diff
@@ -10,9 +10,29 @@ const (
 	HashLength   = 32
 )
 
-// CidBuilder is the default CID builder for Filecoin.
-var CidBuilder cid.Builder = cid.V1Builder{
-	Codec:    cid.DagCBOR,
-	MhLength: HashLength,
-	MhType:   HashFunction,
+type cidBuilder struct {
+	codec uint64
+}
+
+func (cidBuilder) WithCodec(c uint64) cid.Builder {
+	return cidBuilder{codec: c}
+}
+
+func (b cidBuilder) GetCodec() uint64 {
+	return b.codec
 }
+
+func (b cidBuilder) Sum(data []byte) (cid.Cid, error) {
+	hf := HashFunction
+	if len(data) <= HashLength {
+		hf = mh.IDENTITY
+	}
+	return cid.V1Builder{Codec: b.codec, MhType: hf}.Sum(data)
+}
+
+// CidBuilder is the default CID builder for Filecoin.
+//
+// - The default codec is CBOR. This can be changed with CidBuilder.WithCodec.
+// - The default hash function is 256bit blake2b when the data is > 32 bytes
```

Comment:
You are changing CIDs across the board with this PR. Doesn't it make sense to ditch `blake2b` as well, since it was selected before `sha2-ni` was a thing?

Reply:
It might. Thoughts @whyrusleeping?

Reply:
@whyrusleeping @Stebalien If nothing else this will shave 2 bytes from every CID:
[`0xA0 0xE4 0x02`](https://gchq.github.io/CyberChef/#recipe=From_Base(16)VarInt_Encode()To_Hex('Space',0)&input=MHhiMjIwCgo) vs `0x12`